pkgs/development/python-modules/watchdog/default.nix

```diff
@@ -43,6 +43,12 @@ buildPythonPackage rec {
       --replace "--cov-report=term-missing" ""
   '';
 
+  # otherwise "tests/test_inotify_c.py::test_select_fd"
+  # fails with "OSError: [Errno 24] Too many open files"
+  preCheck = ''
+    ulimit -n 4096
```

Comment:
Do the other tests open at most 2048 files?

Reply:
Well, it's actually a bit more than 2048, since `pytest` itself opens files too, but yes, it's the only test that opens that many files, AFAICS.